csharp/src/Google.Protobuf/Reflection/Dynamic/DynamicMessage.cs

```diff
@@ -0,0 +1,377 @@
+                }
+                throw new Exception("FieldType:" + fieldType + ", not handled in GetFieldCodec method.");
+            }
+
+            /// <summary>
+            /// Reads the field from the CodedInputStream, based on the fieldType and returns it.
+            /// </summary>
+            /// <param name="fieldType"></param>
+            /// <param name="input"></param>
+            /// <returns></returns>
+            private object ReadField(FieldType fieldType, CodedInputStream input)
+            {
+                switch (fieldType)
+                {
+                    case FieldType.Int32:
+                        return input.ReadInt32();
+                    case FieldType.Int64:
+                        return input.ReadInt64();
+                    case FieldType.Bytes:
+                        return input.ReadBytes();
+                    case FieldType.String:
+                        return input.ReadString();
+                    case FieldType.Double:
+                        return input.ReadDouble();
+                    case FieldType.Float:
+                        return input.ReadFloat();
+                    case FieldType.Bool:
+                        return input.ReadBool();
+                    case FieldType.UInt32:
+                        return input.ReadUInt32();
+                    case FieldType.UInt64:
+                        return input.ReadUInt64();
+                    case FieldType.Fixed32:
+                        return input.ReadFixed32();
+                    case FieldType.Fixed64:
+                        return input.ReadFixed64();
+                    case FieldType.SFixed64:
+                        return input.ReadSFixed64();
+                    case FieldType.SFixed32:
+                        return input.ReadSFixed32();
+                    case FieldType.SInt32:
+                        return input.ReadSInt32();
+                    case FieldType.SInt64:
+                        return input.ReadSInt64();
+                    case FieldType.Enum:
+                        return input.ReadEnum();
+                }
+                throw new Exception("FieldType:" + fieldType + ", not handled while reading field.");
+            }
+
+            /// <summary>
+            /// Marks the fields as immutable and returns a newly created DynsmicMessage.
+            /// </summary>
+            /// <returns></returns>
+            internal DynamicMessage BuildParsed()
+            {
+                DynamicMessage result = new DynamicMessage(type, fields);
+                return result;
+            }
+
+            /// <summary>
+            /// Default implementation of Overridden method.
+            /// </summary>
+            /// <param name="output"></param>
+            /// <exception cref="NotImplementedException"></exception>
+            void IMessage.WriteTo(CodedOutputStream output)
+            {
+                throw new NotImplementedException();
```

Comment:
If this is actually not needed, it would be good to have a 1-liner explaining why, same with the method below

Reply:
Done